internal/planner/mapper/select.go

```diff
@@ -38,6 +38,10 @@ type Select struct {
 	// These can include stuff such as version information, aggregates, and other
 	// Selects.
 	Fields []Requestable
+
+	// SkipResolve is a flag that indicates that the fields in this Select don't need to be resolved.
+	// It is used to avoid resolving related objects if they are used only in a filter and not requested in a response.
+	SkipResolve bool
```

Comment:
suggestion: It took a bit of head-scratching and reading to be sure, but I think I understand what you mean by resolve - however the term (within this context) was new to me, and the comment doesn't define what it means.

I think adding an explanation of that to this comment could be quite handy in the future.

Reply:
I thought it was a common term in graphql space which refers to a resolver that is supposed to set a value for a requested object field.
If you have a better suggestion, let me know please. Otherwise I will just extend the documentation.

Reply:
I'm happy with the current name, documentation would be nice though - thanks :)

Reply:
done